### PR TITLE
chore: adding upgrade states to channel state enum

### PR DESCRIPTION
**Pull request by chatton**
## Description

closes: #1604

---

Before we can merge this PR, please make sure that all the following items have been
checked off. If any of the checklist items are not applicable, please leave them but
write a little note why.

- [ ] Targeted PR against correct branch (see [CONTRIBUTING.md](https://github.com/cosmos/ibc-go/blob/master/CONTRIBUTING.md#pr-targeting))
- [ ] Linked to Github issue with discussion and accepted design OR link to spec that describes this work.
- [ ] Code follows the [module structure standards](https://github.com/cosmos/cosmos-sdk/blob/master/docs/building-modules/structure.md).
- [ ] Wrote unit and integration [tests](https://github.com/cosmos/ibc-go/blob/master/CONTRIBUTING.md#testing)
- [ ] Updated relevant documentation (`docs/`) or specification (`x/<module>/spec/`)
- [ ] Added relevant `godoc` [comments](https://blog.golang.org/godoc-documenting-go-code).
- [ ] Added a relevant changelog entry to the `Unreleased` section in `CHANGELOG.md`
- [ ] Re-reviewed `Files changed` in the Github PR explorer
- [ ] Review `Codecov Report` in the comment section below once CI passes

### CODE REVIEW

File: proto/ibc/core/channel/v1/channel.proto [hunk 1]
```diff
@@ -65,6 +65,12 @@ enum State {
   // A channel has been closed and can no longer be used to send or receive
   // packets.
   STATE_CLOSED = 4 [(gogoproto.enumvalue_customname) = "CLOSED"];
+  // A channel has just started the channel upgrade handshake.
+  // The chain that is proposing the upgrade should set the channel state from OPEN to UPGRADE_INIT.
+  STATE_UPGRADE_INIT = 5 [(gogoproto.enumvalue_customname) = "UPGRADE_INIT"];
+  // A channel has acknowledged the upgrade handshake step on the counterparty chain.
+  // The counterparty chain that accepts the upgrade should set the channel state from OPEN to UPGRADE_TRY.
+  STATE_UPGRADE_TRY = 6 [(gogoproto.enumvalue_customname) = "UPGRADE_TRY"];
```
Comment (damiannolan) [inline proto/ibc/core/channel/v1/channel.proto:73]: Protobuf defined enums are perfect! Maybe we should remove the underscore between "UPGRADE" and "INIT"/"TRY".

See here for the revised updates to what we should use for `gogoproto.enumvalue_customname`

https://github.com/cosmos/ibc/pull/803

This is essentially to align with the same naming convention used for `STATE_TRYOPEN` / `TRYOPEN` enum.

**Review by damiannolan — CHANGES_REQUESTED** [review]

File: proto/ibc/core/channel/v1/channel.proto [hunk 2]
```diff
@@ -65,6 +65,12 @@ enum State {
   // A channel has been closed and can no longer be used to send or receive
   // packets.
   STATE_CLOSED = 4 [(gogoproto.enumvalue_customname) = "CLOSED"];
+  // A channel has just started the channel upgrade handshake.
+  // The chain that is proposing the upgrade should set the channel state from OPEN to UPGRADEINIT.
+  STATE_UPGRADEINIT = 5 [(gogoproto.enumvalue_customname) = "UPGRADEINIT"];
```
Comment (damiannolan) [inline proto/ibc/core/channel/v1/channel.proto:70]: I think we should go for `INITUPGRADE` and `TRYUPGRADE` to align with the `TRYOPEN` state.
Apologies for the nit picking 😆

Comment (chatton) [reply proto/ibc/core/channel/v1/channel.proto:70]: ah you're right makes total sense!

**Review by damiannolan — APPROVED** [review]
Cheers @chatton! LGTM 🚀

**Pull request merged**